src/main/java/com/powsybl/openloadflow/ac/MonitoringVoltageOuterLoop.java

```diff
@@ -0,0 +1,158 @@
+/**
+ * Copyright (c) 2022, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+package com.powsybl.openloadflow.ac;
+
+import com.powsybl.commons.reporter.Reporter;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoop;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoopContext;
+import com.powsybl.openloadflow.ac.outerloop.OuterLoopStatus;
+import com.powsybl.openloadflow.network.LfBus;
+import com.powsybl.openloadflow.network.LfGenerator;
+import com.powsybl.openloadflow.network.LfStaticVarCompensator;
+import com.powsybl.openloadflow.util.Reports;
+import org.apache.commons.lang3.Range;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * @author Anne Tilloy <anne.tilloy at rte-france.com>
+ */
+public class MonitoringVoltageOuterLoop implements OuterLoop {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(MonitoringVoltageOuterLoop.class);
+
+    @Override
+    public String getType() {
+        return "Voltage monitoring";
+    }
+
+    private enum VoltageLimitDirection {
+        MIN,
+        MAX
+    }
+
+    private static final class PqToPvBus {
+
+        private final LfBus controllerBus;
+
+        private final VoltageLimitDirection voltageLimitDirection;
+
+        private PqToPvBus(LfBus controllerBus, VoltageLimitDirection voltageLimitDirection) {
+            this.controllerBus = controllerBus;
+            this.voltageLimitDirection = voltageLimitDirection;
+        }
+    }
+
+    @Override
+    public void initialize(OuterLoopContext context) {
+        for (LfBus bus : context.getNetwork().getBuses()) {
+            bus.setVoltageControlSwitchOffCount(0);
```

Comment:
We don't need switch off counter for SVC?

Reply:
Yes, you are right because it can become a classical PV -> PQ -> PV bus after.

Reply:
Indeed it is done by the reactive limits outerloop so I can delete this.